presto-main/src/test/java/com/facebook/presto/cost/TestCostCalculator.java

```diff
@@ -527,11 +531,14 @@ private CostAssertionBuilder assertCostFragmentedPlan(
             Map<String, PlanNodeStatsEstimate> stats,
             Map<String, Type> types)
     {
-        TypeProvider typeProvider = TypeProvider.copyOf(types.entrySet().stream()
-                .collect(ImmutableMap.toImmutableMap(entry -> new Symbol(entry.getKey()), Map.Entry::getValue)));
+        Map<Symbol, Type> symbolTypes = types.entrySet().stream()
+                .collect(ImmutableMap.toImmutableMap(entry -> new Symbol(entry.getKey()), Map.Entry::getValue));
+        TypeProvider typeProvider = TypeProvider.copyOf(symbolTypes);
         StatsProvider statsProvider = new CachingStatsProvider(statsCalculator(stats), session, typeProvider);
         CostProvider costProvider = new TestingCostProvider(costs, costCalculatorUsingExchanges, statsProvider, session, typeProvider);
-        SubPlan subPlan = fragment(new Plan(node, typeProvider, StatsAndCosts.create(node, statsProvider, costProvider)));
+        IterativeOptimizer optimizer = new IterativeOptimizer(new RuleStatsRecorder(), statsCalculator(stats), costCalculatorUsingExchanges, new TranslateExpressions(metadata, new SqlParser()).rules());
+        PlanNode plan = optimizer.optimize(node, session, typeProvider, new SymbolAllocator(symbolTypes), new PlanNodeIdAllocator(), WarningCollector.NOOP);
```

Comment:
Why we need to do this before fragmenting?

Reply:
Because, from fragmented plan, we only expects `RowExpression`.

Reply:
Actually, it is weird, after rebase and squash, I no longer receive any error in `Fragmenter `...  I can remove this whole commit.

Reply:
It is because the Fragmenter called PlanPrinter...